source/math/FloatBox2D.ooc

```diff
@@ -63,6 +63,18 @@ FloatBox2D: cover {
 	shrink: func ~fraction (margin: Float) -> This {
 		this pad(-margin * this height / 2.0f)
 	}
+	scale: func (value: Float) -> This {
+		sizeChange := this size - (this size * value)
+		This new(this leftTop + (sizeChange / 2.0f), this size - sizeChange)
+	}
+	enlarge: func (size: FloatSize2D) -> This {
+		sizeChange := (size - this size) maximum~Float(0.0f)
+		This new(this leftTop - (sizeChange / 2.0f), this size + sizeChange)
+	}
+	reduce: func (size: FloatSize2D) -> This {
+		sizeChange := (size - this size) minimum~Float(0.0f)
+		This new(this leftTop - (sizeChange / 2.0f), this size + sizeChange)
+	}
```

Comment:
`reduce` could mean either `reduceTo` or `reduceBy`. How about `enlargeTo` and `shrinkTo`? Or are there even better names?